services/room_cancelation/auto_cancel_event.py

```diff
@@ -0,0 +1,131 @@
+from dateutil.relativedelta import relativedelta
+from api.room.models import Room as RoomModel
+from api.events.models import Events as EventsModel
+from datetime import datetime
+from helpers.calendar.credentials import Credentials
+
+
+class UpdateRecurringEvent():
+    """
+    This class is used:
+    - to get recurring events from calendar api
+    - update them in the events model
+    - cancel any room booking for events that have not
+    been checked into three consecutive times
+
+    To the functionality of this class run the command
+    'python run_autocancel_room.py' from the root folder
+    """
+    def get_room_index_from_attendees(self, attendees, calendar_id):
+        """
+        :param attendees: this a list of attendees of an event
+        as returned by the Calendar API
+        :param calendar_id: This is the id calendar_id of the
+        room whose index you need to get
+        :return: The index of the room in the list of attendees
+        """
+        index_of_room = 0
+        for index in range(len(attendees)):
+            if attendees[index]["email"] == calendar_id:
+                index_of_room = index
+                break
+        return index_of_room
+
+    def get_all_recurring_events(self, query, start_date, end_date):
+        """
+        :param query: A room query that allows you to query all
+        the active rooms
+        :param start_date: the first day from which you want to get
+        the recurring events
+        :param end_date: the final date upto which you want to get
+        the recurring event
+        :return: the recurring events within the time frame
+        """
+        rooms = query.filter_by(state="active")
+        credentials = Credentials()
+        service = credentials.set_api_credentials()
+        recurring_events = []
+        for room in rooms:
+            events = []
+            try:
+                events = service.events().list(
+                    calendarId=room.calendar_id, timeMax=end_date,
+                    timeMin=start_date, singleEvents=True,
+                    orderBy='startTime').execute()
+            except Exception:
+                continue
+            for event in events['items']:
+                recurring_event_id = event.get("recurringEventId")
+                if not event.get("attendees"):
+                    continue
+                room_index = self.get_room_index_from_attendees(
+                    event["attendees"],
+                    room.calendar_id
+                )
+                room_response_status = event.get("attendees")[room_index]["responseStatus"]  # noqa: E501
```

Comment:
Is it possible for us to do away with this `#noqa`?

Reply:
Yeah, on it. Thanks @nakatuddesuzan